### PR TITLE
About the base

### PR DESCRIPTION
**Pull request by veronika-kolosova**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by solomiyon — APPROVED** [review]

**Review by OleksandrBilan — APPROVED** [review]

**Review by andrii52 — APPROVED** [review]

**Review by danylo-melnychuk — APPROVED** [review]

File: EPlast/EPlast.BLL/DTO/AboutBase/SubsectionDTO.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+namespace EPlast.BLL.DTO.AboutBase
+{
+    public class SubsectionDTO
+    {
+        public int Id { get; set; }
+        public string Title { get; set; }
+
+        public int SectionId { get; set; }
+        public SectionDTO Section { get; set; }
+
```
Comment (voroniak) [inline EPlast/EPlast.BLL/DTO/AboutBase/SubsectionDTO.cs:14]: delete spaces

File: EPlast/EPlast.BLL/DTO/AboutBase/SubsectionDTO.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace EPlast.BLL.DTO.AboutBase
+{
+    public class SubsectionDTO
+    {
+        public int Id { get; set; }
+        public string Title { get; set; }
+
```
Comment (voroniak) [inline EPlast/EPlast.BLL/DTO/AboutBase/SubsectionDTO.cs:11]: delete spaces

File: EPlast/EPlast.BLL/DTO/AboutBase/SubsectionDTO.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        public int SectionId { get; set; }
+        public SectionDTO Section { get; set; }
+
+        public string Description { get; set; }
+
```
Comment (voroniak) [inline EPlast/EPlast.BLL/DTO/AboutBase/SubsectionDTO.cs:16]: delete spaces

File: EPlast/EPlast.BLL/Interfaces/AboutBase/IAboutBaseSectionService.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using EPlast.BLL.DTO.AboutBase;
+using EPlast.DataAccess.Entities;
+
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Interfaces/AboutBase/IAboutBaseSectionService.cs:5]: delete spaces

File: EPlast/EPlast.BLL/Mapping/AboutBase/AboutBaseSectionProfile.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+namespace EPlast.BLL.Mapping.AboutBase
+{
+    class AboutBaseSectionProfile: Profile
+    {
+        public AboutBaseSectionProfile()
+        {
+            CreateMap<Section, SectionDTO>().ReverseMap();
+        }
+    }
+}
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Mapping/AboutBase/AboutBaseSectionProfile.cs:14]: format file. crtl r g

File: EPlast/EPlast.BLL/Mapping/AboutBase/AboutBaseSubsectionProfile.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+namespace EPlast.BLL.Mapping.AboutBase
+{
+    class AboutBaseSubsectionProfile:Profile
+    {
+        public AboutBaseSubsectionProfile()
+        {
+            CreateMap<Subsection, SubsectionDTO>().ReverseMap();
+        }
+    }
+}
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Mapping/AboutBase/AboutBaseSubsectionProfile.cs:14]: format file. crtl r g

File: EPlast/EPlast.BLL/Services/AboutBase/AboutBaseSectionService.cs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace EPlast.BLL.Services.AboutBase
+{
+    public class AboutBaseSectionService : IAboutBaseSectionService
+    {
+        private readonly IRepositoryWrapper _repoWrapper;
+        private readonly IMapper _mapper;
+
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Services/AboutBase/AboutBaseSectionService.cs:18]: delete spaces

File: EPlast/EPlast.BLL/Services/AboutBase/AboutBaseSubsectionService.cs [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        private readonly IMapper _mapper;
+        private readonly UserManager<User> _userManager;
+
+        public AboutBaseSubsectionService(IRepositoryWrapper repositoryWrapper, IMapper mapper, UserManager<User> userManager)
+        {
+            _repoWrapper = repositoryWrapper;
+            _mapper = mapper;
+            _userManager = userManager;
+        }
+
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Services/AboutBase/AboutBaseSubsectionService.cs:26]: delete spaces. The same for other files. Delete unneeded spaces and format files.

File: EPlast/EPlast.Tests/Controllers/AboutBaseControllerTests.cs [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+            var result = await _aboutbaseController.GetAboutBaseSubsection(It.IsAny<int>());
+            var resultValue = (result as OkObjectResult).Value;
+            //Assert
+            _subsectionSercive.Verify();
+            Assert.IsNotNull(result);
+            Assert.IsInstanceOf<OkObjectResult>(result);
+            Assert.IsNotNull(resultValue);
+            Assert.IsInstanceOf<SubsectionDTO>(resultValue);
+        }
+        [Test]
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Controllers/AboutBaseControllerTests.cs:109]: add space. The same for all tests.

File: EPlast/EPlast.Tests/Services/AboutBase/SectionServiceTests.cs [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+        Section Section = new Section
+        {
+            Id = 1,
+            Title = "Title"
+        };
+        SectionDTO SectionDTO = new SectionDTO 
+        { 
+            Id = 1,
+            Title = "Title"
+        };
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Services/AboutBase/SectionServiceTests.cs:242]: Could you add "private" to this fields?

File: EPlast/EPlast.Tests/Services/AboutBase/SubsectionServiceTests.cs [hunk 11]
```diff
@@ -0,0 +1,283 @@
...
+            mockRepoWrapper
+               .Setup(x => x.AboutBaseSubsection.GetFirstAsync(It.IsAny<Expression<Func<Subsection, bool>>>(),
+                   It.IsAny<Func<IQueryable<Subsection>, IIncludableQueryable<Subsection, object>>>()))
+               .ReturnsAsync(Subsection);
+            userManager.Setup(m => m.GetRolesAsync(It.IsAny<User>())).ReturnsAsync(GetRolesWithoutAdmin());
+            //Assert
+
+            Exception exception = Assert.ThrowsAsync(typeof(UnauthorizedAccessException),
+                async () => { await AboutBaseSubsectionService.DeleteSubsection(It.IsAny<int>(), It.IsAny<User>()); });
+            Assert.AreEqual("Attempted to perform an unauthorized operation.", exception.Message);
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Services/AboutBase/SubsectionServiceTests.cs:157]: You use "Attempted to perform an unauthorized operation." several times. isn't it better to put it in a variable or private field?

File: EPlast/EPlast.Tests/Services/AboutBase/SubsectionServiceTests.cs [hunk 12]
```diff
@@ -0,0 +1,283 @@
...
+        Subsection Subsection = new Subsection
+        {
+            Id = 1,
+            Title = "Title"
+        };
+        SubsectionDTO SubsectionDTO = new SubsectionDTO
+        {
+            Id = 1,
+            Title = "Title"
+        };
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Services/AboutBase/SubsectionServiceTests.cs:242]: add private

File: EPlast/EPlast.WebApi/Controllers/AboutBaseController.cs [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+        {
+            try
+            {
+                await _aboutBaseSectionService.DeleteSection(id, await _userManager.GetUserAsync(User));
+                return NoContent();
+            }
+            catch (NullReferenceException)
+            {
+                return NotFound();
+            }
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/AboutBaseController.cs:81]: I don't think that using try catch so many time is ok. don't we have global error handling?

File: EPlast/EPlast.WebApi/Mapping/AboutBase/AboutBaseSectionProfile.cs [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+using EPlast.BLL.DTO.AboutBase;
+using EPlast.WebApi.Models.AboutBase;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace EPlast.WebApi.Mapping.AboutBase
+{
+    public class AboutBaseSectionProfile: Profile
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Mapping/AboutBase/AboutBaseSectionProfile.cs:11]: Why did you create some Profiles public but the others no?

**Review by voroniak — CHANGES_REQUESTED** [review]

**Review by nicoolle — APPROVED** [review]

**Review by voroniak — APPROVED** [review]

**Pull request merged**